### ISSUE TITLE
onaudioprocess events after finish

### ISSUE THREAD

**Issue by manelj** [higuma/web-audio-recorder-js#8]
MP3 Encoder
I have this error: TypeError: recBuffers is undefined

Inspecting the code i have found that when i finish recording
`audioRecorder.finishRecording();`
computer continues processing this events
`this.processor.onaudioprocess = function(event) {
          for (var ch = 0; ch < numChannels; ++ch)
            buffer[ch] = event.inputBuffer.getChannelData(ch);
          worker.postMessage({ command: "record", buffer: buffer });
manel++;
console.log("manel " + manel);
        };
`
after the cleanup function was called here in WebAudioRecorderMp3.js:
`
function finish() {
  if (recBuffers) {
    postProgress(0);
    encoder = new Mp3LameEncoder(sampleRate, options.mp3.bitRate);
    var timeout = Date.now() + options.progressInterval;
    while (recBuffers.length > 0) {
      encoder.encode(recBuffers.shift());
      var now = Date.now();
      if (now > timeout) {
        postProgress((bufferCount - recBuffers.length) / bufferCount);
        timeout = now + options.progressInterval;
      }
    }
    postProgress(1);
  }
  self.postMessage({
    command: "complete",
    blob: encoder.finish(options.mp3.mimeType)
  });
  cleanup();
};
`

and then the error throws here in WebAudioRecorderMp3.js, because encoder now is null
`
function record(buffer) {
  if (bufferCount++ < maxBuffers)
    if (encoder)
      encoder.encode(buffer);
    else
      recBuffers.push(buffer);
  else
    self.postMessage({ command: "timeout" });
};
`

**Comment by salkj** [comment]
Hi manelj, 

Did you ever figure this out? I currently am suffering from the same TypeError.

**Comment by kaushik-evani** [comment]
Yes I'm facing the same issue. And btw, I'm only encountering this on Firefox. It works fine on chrome. And this also happens with WAV encoding for me. Any solutions to this please?
